### PR TITLE
feat: implement ResponseWrapperMiddleware

### PR DESCRIPTION
**Pull request by xuanli520**
feat: implement unified response model and middleware

## Related Issue

#24 

## Summary of Changes

Implements a ResponseWrapperMiddleware:

  Core Components:
  - Response Model (src/responses/base.py): Generic Response[T] class with success() and error() factory methods,
  providing consistent API response structure
  - UnifiedResponseMiddleware (src/responses/middleware.py): ASGI middleware that automatically wraps all JSON
  responses into the unified format, handles error extraction, and preserves headers/cookies

  Key Features:
  - Automatic response wrapping for all API endpoints (200-399 success, 400+ errors)
  - Smart skipping of documentation paths (/docs, /redoc, /openapi.json)
  - Detection and preservation of already-wrapped responses (no double-wrapping)
  - Error message extraction from detail, msg, or message fields
  - Header and cookie preservation across wrapped responses
  - Support for all JSON-serializable data types (dict, list, string, None)

  Testing:
  - 499 lines of comprehensive tests covering unit, integration, and edge cases
  - Tests for middleware behavior, error handling, header preservation, and response structure validation

## Breaking Changes

N/A

## Checklist

- [x] Issue discussion completed before opening PR
- [x] Scope is small and focused (single feature/fix)
- [x] All functions have full type annotations
- [x] Async/await used for all I/O operations
- [x] Tests added for new behaviors
- [x] `uv run pre-commit run --all-files` passes
- [x] `uv run pytest` passes

### CODE REVIEW

**Comment by xuanli520** [comment]
image

File: src/responses/middleware.py [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+        unified = JSONResponse(
+            content=new_payload,
+            status_code=200,
+            media_type=api_response.media_type,
+            background=api_response.background,
+        )
+        self._copy_headers(api_response, unified)
+        return unified
+
+    def _should_skip(self, request: Request, response: FastAPIResponse) -> bool:
```
Comment (Besthope-Official) [inline src/responses/middleware.py]: 第二个检查是多余的

File: src/responses/middleware.py [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+            if detail is not None:
+                return json.dumps(detail, ensure_ascii=False)
+
+        if isinstance(payload, list):
+            return json.dumps(payload, ensure_ascii=False)
+
+        return "error"
+
+    @staticmethod
+    def _copy_headers(source: FastAPIResponse, target: JSONResponse) -> None:
```
Comment (Besthope-Official) [inline src/responses/middleware.py]: 完全不必要，JsonResponse 应该已经能够自己设置header，删掉

File: src/responses/middleware.py [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+from fastapi import Request
+from fastapi.responses import JSONResponse, Response as FastAPIResponse
+from starlette.concurrency import iterate_in_threadpool
+from starlette.middleware.base import BaseHTTPMiddleware
+from starlette.types import ASGIApp
+
+from .base import Response
+
+
+class UnifiedResponseMiddleware(BaseHTTPMiddleware):
```
Comment (Besthope-Official) [inline src/responses/middleware.py]: 改个名字吧，这个“统一响应中间件”不是很清晰，换成ResponseWrapperMiddleware，表示对 response warp

**Review by Besthope-Official — CHANGES_REQUESTED** [review]
自审中间件的部分代码是否有重复造轮的嫌疑
